pkg/ref/ref.go

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2022. Nydus Developers. All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package ref
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path"
+
+	"github.com/opencontainers/go-digest"
+	"github.com/pkg/errors"
+
+	"github.com/containerd/containerd/content/local"
+	"github.com/containerd/nydus-snapshotter/pkg/converter"
+	"github.com/containerd/nydus-snapshotter/pkg/label"
+	"github.com/containerd/nydus-snapshotter/pkg/resolve"
+	"github.com/containerd/nydus-snapshotter/pkg/utils/registry"
+)
+
+type Manager struct {
+	blobCacheDir string
+	resolver     *resolve.Resolver
+}
+
+func NewManager(blobCacheDir string, resolver *resolve.Resolver) *Manager {
+	return &Manager{
+		blobCacheDir: blobCacheDir,
+		resolver:     resolver,
+	}
+}
+
+func (m *Manager) PrepareRefLayer(labels map[string]string) error {
+	ref, layerDigest := registry.ParseLabels(labels)
+	if ref == "" || layerDigest == "" {
+		return fmt.Errorf("can't find ref and digest from label %+v", labels)
+	}
+
+	rc, err := m.resolver.Resolve(ref, layerDigest, labels)
+	if err != nil {
+		return errors.Wrapf(err, "resolve from ref %s, digest %s", ref, layerDigest)
+	}
+	defer rc.Close()
+
+	blobFile, err := os.CreateTemp(m.blobCacheDir, "downloading-")
```

Comment:
I hope snapshotter does not deal with any layer downloads, which I think can be delagated to nydusd